conans/client/conan_api.py

```diff
@@ -544,24 +539,67 @@ def package(self, reference="", package_id=None, build_folder=None, source_folde
             self._manager.local_package(package_folder, recipe_folder, build_folder, source_folder)
 
     @api_method
-    def source(self, reference, force=False, cwd=None):
-        cwd = prepare_cwd(cwd)
-        current_path, reference = _get_reference(reference, cwd)
-        self._manager.source(current_path, reference, force)
+    def source(self, path, source_folder=None, build_folder=None):
+        path = self._abs_relative_to(path, os.getcwd())
+        source_folder = self._abs_relative_to(source_folder, os.getcwd()) or os.getcwd()
```

Comment:
A bit ugly so much ``os.getcwd()``. Maybe add an argument to the ``abs_relative_to``, that is defaulted to None, and could be passed the cwd:

```python
cwd = os.getcwd()
path = self._abs_relative_to(path, cwd)
source_folder = self._abs_relative_to(source_folder, cwd, default_dir=cwd)
```